src/Library/DeviceBase.cpp

```diff
@@ -90,14 +90,20 @@ namespace usbguard
      * Set connect type
      */
     int retry = 0;
-    std::string connect_type;
-    while (retry < 20) {
+    std::string connect_type = "";
+    /*
+     * Read port/connect_type only for non Host Controller devices
+     */
+    if (getPort().substr(0, 3) != "usb") {
```

Comment:
My vote for turning this into
- a) `hasPrefix(getPort(), "usb")` plus `#include "Common/Utility.hpp"`
- b) `isController()` (but but does more than just checking the port, please double check if that's in our interest, I'm unsure)

Reply:
I would go with option "a"

I have tried using isController() method but the additional information that it is checking more than port is not available in the sysfs at this point of code. The descriptors are parsed later the connect_type is set and host controllers can't be filtered using isController() here. That additional check is out of our interest here.

Reply:
Sounds good!